pkg/controller/servicemesh/controlplane/controller.go

```diff
@@ -255,6 +255,15 @@ func (r *ReconcileControlPlane) Reconcile(request reconcile.Request) (reconcile.
 		reqLogger.V(1).Info("Adding finalizer", "finalizer", common.FinalizerName)
 		finalizers.Insert(common.FinalizerName)
 		instance.SetFinalizers(finalizers.List())
+		// set default version if necessary
+		if len(instance.Spec.Version) == 0 {
+			if len(instance.Status.AppliedVersion) == 0 {
+				instance.Spec.Version = common.DefaultMaistraVersion
```

Comment:
Should we really write to the `Spec`? I think this should go into `Status`

Reply:
Good question.  My rationale was that version selection is an explicit action and if the user doesn't specify the version, we should initialize it with the default, to help prevent unwanted upgrades in the future.  If the field is empty, it is assumed that the version is 'v1.0', as the field did not exist in the previous version of the operator.

Reply:
We should probably converge on a single place to specify the version. Maybe you're right and it should be `Spec.Version`, in that case I'd suggest we make that the full version though (`1.1.0` instead of `v1.1`) and derive the short form from that (we know that 1.1.0 needs v1.1 charts) - but also give users the possibility to specify `1.1` and we will select the latest `1.1.x` release for them. We can then populate the `global.tag` field based on that choice

Reply:
But I still think we should only write to `Status` ;) Maybe @luksa can enlighten us

Reply:
I like the idea of using the full version, however there could be chart differences between .z releases which would mean a downgrade of the operator might get you 1.1.0 charts with a 1.0.2 tag.  I'm not sure if that's an issue or not, just saying.

Reply:
I actually made a comment on this yesteday, but it seems to have disappeared. Weird.

Yes, I'm not a fan of controllers updating the object's spec. It always causes problems. Users should be the only ones modifying it. For example, what if the user applies a modified SMCP after the controller has modified the version. Does the version get reset to "" again?

Defaulting, which is what the code here is doing, is usually implemented in admission webhooks. If the user leaves the version empty, it should be defaulted to the current version. But any existing SMCPs without an empty version should be assumed to be v1.0. The controller should handle this without modifying the SMCP. 

Actually, I'm not sure this code here even does what it should. The object's status is always empty when this code is invoked (when the finalizer isn't there yet), so it will always set the default maistra version. Plus, this is never invoked if the finalizer is already set (e.g. on existing SMCPs).

Reply:
I'll fix this.